### DISCUSSION TITLE
Type never for Sprinkles with shorthands

### DISCUSSION THREAD

**Discussion by d-kls** [vanilla-extract-css/vanilla-extract#828] [Q&A]
I have the following basic test setup.

```typescript
const someTestProperties = defineProperties({
	properties: {
		paddingTop: {
			sm: "12px",
			md: "14px",
		},
		paddingBottom: {
			sm: "12px",
			md: "14px",
		},
	},
	shorthands: {
		py: ["paddingTop", "paddingBottom"],
	},
});
const testAtoms = createSprinkles(someTestProperties);
type TypeTest = Parameters<typeof testAtoms>[0];
```

TypeTest will show here as:
![image](https://user-images.githubusercontent.com/68340074/190171936-3371d9ad-61cb-4337-9c01-9b1d37d1462e.png)

and i don't know why - since this simple example should work afaik. As soon as i remove the shorthands i get autocomplete and suggestion for my type again. I must be missing something surely...

UPDATE:
Seems to be solved now - i was checking the TypeScript version of the TypeScript service used in the IDE and it was below 4. Perhaps that caused the issue (?) - after changing it and restarting it worked.

**Comment by hauptrolle** [comment]
@markdalgleish Do you have any updates on this? This happens to me also, only in Webstorm somehow 🤔
